spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestKeyedTableDML.java

```diff
@@ -95,4 +111,42 @@ public void testSelectChangeFiles() {
     Assert.assertEquals(2, rows.size());
   }
 
+
+  @Test
+  public void testSelectDeleteTable() throws IOException {
+    dataFiles = writeBase(TableIdentifier.of(catalogName, database, table), baseFiles);
+    insertBasePosDeleteFiles(2);
+    rows = sql("select * from {0}.{1}", database, table);
+    Assert.assertEquals(0, rows.size());
+  }
+
+  protected List<DeleteFile> insertBasePosDeleteFiles(long transactionId) throws IOException {
+    Map<StructLike, List<DataFile>> dataFilesPartitionMap =
+        new HashMap<>(dataFiles.stream().collect(Collectors.groupingBy(ContentFile::partition)));
+    List<DeleteFile> deleteFiles = new ArrayList<>();
+    for (Map.Entry<StructLike, List<DataFile>> dataFilePartitionMap : dataFilesPartitionMap.entrySet()) {
+      StructLike partition = dataFilePartitionMap.getKey();
+      List<DataFile> partitionFiles = dataFilePartitionMap.getValue();
+      Map<DataTreeNode, List<DataFile>> nodeFilesPartitionMap = new HashMap<>(partitionFiles.stream()
+          .collect(Collectors.groupingBy(dataFile ->
+              DefaultKeyedFile.parseMetaFromFileName(dataFile.path().toString()).node())));
+      for (Map.Entry<DataTreeNode, List<DataFile>> nodeFilePartitionMap : nodeFilesPartitionMap.entrySet()) {
+        DataTreeNode key = nodeFilePartitionMap.getKey();
+        List<DataFile> nodeFiles = nodeFilePartitionMap.getValue();
+
+        // write pos delete
+        SortedPosDeleteWriter<Record> writer = GenericTaskWriters.builderFor(keyedTable)
+            .withTransactionId(transactionId).buildBasePosDeleteWriter(key.getMask(), key.getIndex(), partition);
+        for (DataFile nodeFile : nodeFiles) {
+          writer.delete(nodeFile.path(), 0);
+        }
+        deleteFiles.addAll(writer.complete());
+      }
+    }
+    RowDelta rowDelta = keyedTable.baseTable().newRowDelta();
+    deleteFiles.forEach(rowDelta::addDeletes);
+    rowDelta.commit();
```

Comment:
delete partial  records in data file， not delete all data in file.  

do not  just test the most simple case. 

test with delete partial records in some data file, and some total data file delete, both case is required

Reply:
result check shoud check record is right